src/rp2040_arduino_platform.cpp

```diff
@@ -91,6 +155,73 @@ void RP2040ArduinoPlatform::commitToEeprom()
 {
     EEPROM.commit();
 }
+
+#endif
+
+#else
+
+size_t RP2040ArduinoPlatform::flashEraseBlockSize()
+{
+    return 16; // 16 pages x 256byte/page = 4096byte
+}
+
+size_t RP2040ArduinoPlatform::flashPageSize()
+{
+    return 256;
+}
+
+uint8_t* RP2040ArduinoPlatform::userFlashStart()
+{
+    return (uint8_t*)XIP_BASE + KNX_FLASH_OFFSET;
+}
+
+size_t RP2040ArduinoPlatform::userFlashSizeEraseBlocks()
+{
+    if(KNX_FLASH_SIZE <= 0)
+        return 0;
+    else
+        return ( (KNX_FLASH_SIZE - 1) / (flashPageSize() * flashEraseBlockSize())) + 1;
+}
+
+void RP2040ArduinoPlatform::flashErase(uint16_t eraseBlockNum)
+{
+    noInterrupts();
+    rp2040.idleOtherCore();
+
+    flash_range_erase (KNX_FLASH_OFFSET + eraseBlockNum * flashPageSize() * flashEraseBlockSize(), flashPageSize() * flashEraseBlockSize());
+
+    rp2040.resumeOtherCore();
+    interrupts();
+}
+
+void RP2040ArduinoPlatform::flashWritePage(uint16_t pageNumber, uint8_t* data)
+{
+    noInterrupts();
+    rp2040.idleOtherCore();
+
+    flash_range_program(KNX_FLASH_OFFSET + pageNumber * flashPageSize(), data, flashPageSize());
+
+    rp2040.resumeOtherCore();
+    interrupts();
+}
+
+void RP2040ArduinoPlatform::writeBufferedEraseBlock()
+{
+    if(_bufferedEraseblockNumber > -1 && _bufferedEraseblockDirty)
```

Comment:
I think a comment that this is only overwritten because to optimize stuff would be helpfull. Otherwise new platforms will needlessly copy this.

Reply:
the comment is in src/rp2040_arduino_platform.h

Reply:
Sorry I missed that.